reconcile/status_board.py

```diff
@@ -157,82 +309,211 @@ def get_current_products_applications(ocm_api: OCMBaseClient) -> list[Product]:
+                    )
+
+        return desired_abstract_status_board_map
 
-        return_list: list[StatusBoardHandler] = []
-        current_products = {p.name: p for p in current_products_applications}
+    @staticmethod
+    def current_abstract_status_board_map(
+        current_products_applications_services: Iterable[Product],
+    ) -> dict[str, AbstractStatusBoard]:
+        return_value: dict[str, AbstractStatusBoard] = {}
+        for product in current_products_applications_services:
+            return_value[product.name] = product
+            for app in product.applications or []:
+                return_value[f"{product.name}/{app.name}"] = app
+                for service in app.services or []:
+                    return_value[f"{product.name}/{app.name}/{service.name}"] = service
+
+        return return_value
 
-        current_as_mapping: Mapping[str, set[str]] = {
-            c.name: {a.name for a in c.applications or []}
-            for c in current_products_applications
-        }
+    @staticmethod
+    def get_diff(
+        desired_abstract_status_board_map: Mapping[str, AbstractStatusBoard],
+        current_abstract_status_board_map: Mapping[str, AbstractStatusBoard],
+        current_products: Mapping[str, Product],
+    ) -> list[StatusBoardHandler]:
+        return_list: list[StatusBoardHandler] = []
 
         diff_result = diff_mappings(
-            current_as_mapping,
-            desired_product_apps,
+            current_abstract_status_board_map,
+            desired_abstract_status_board_map,
         )
 
-        for product_name in diff_result.add:
-            product = Product(name=product_name, fullname=product_name, applications=[])
+        products_to_create = [
+            p for _, p in diff_result.add.items() if isinstance(p, Product)
+        ]
+        apps_to_create = [
+            a for _, a in diff_result.add.items() if isinstance(a, Application)
+        ]
+        services_to_create = [
+            s for _, s in diff_result.add.items() if isinstance(s, Service)
+        ]
+
+        # Create apps for products that already exists
+        if apps_to_create:
+            for name in current_products:
+                this_product_apps = [
+                    a for a in apps_to_create if a.product and a.product.name == name
+                ]
+                for a in this_product_apps or []:
+                    return_list.append(
+                        StatusBoardHandler(action=Action.create, status_board_object=a)
+                    )
+                    this_app_services = [
+                        s
+                        for s in services_to_create
+                        if s.application and s.application.name == a.name
+                    ]
+                    return_list.extend(
+                        StatusBoardHandler(action=Action.create, status_board_object=s)
+                        for s in this_app_services
+                    )
```

Comment:
why need to handle them differently instead of just

```python
return_list.expand(
    StatusBoardHandler(action=Action.create, status_board_object=o) for o in diff_result.add.values()
)
```

same question for delete

Reply:
Oh, thank you I missed that during refactor.